### ISSUE TITLE
Toegankelijkheidsbevinding Progress Bar: accessible name

### ISSUE THREAD

**Issue by Anto1nette** [dso-toolkit/dso-toolkit#1211]
### Aanleiding
Nu we bezig zijn met storybook, en daarin met de @storybook/addon-a11y add-on toegankelijkheidstests kunnen loslaten op componenten, zien we dat er met de progress-bar iets niet in de haak is: "ARIA progressbar must have an accessible name".

Dit geldt zelfs voor de variant van de progressbar waarin in een <span> extra duiding wordt gegeven:
```
<div class="progress">
  <div
    class="progress-bar"
    role="progressbar"
    aria-valuenow="60"
    aria-valuemin="0"
    aria-valuemax="100"
    style="width: 60%"
  >
    <span>Nog ongeveer 4 minuten</span>
  </div>
</div> 
```

### Werkzaamheden
Axe wijst ons op de mogelijke oplossingen: https://dequeuniversity.com/rules/axe/4.1/aria-progressbar-name?application=axeAPI

We moeten ofwel:
- een aria-labelledby toevoegen (dat zou kunnen voor de variant met label, dan moeten we die span een id geven en daarnaar verwijzen)
- een aria-label toevoegen (die dan aangeeft waar 'progress' op is)
- een title toevoegen (zelfde, maar dan heb je er als 'ziende gebruiker' ook mogelijk nog wat aan)

**Comment by gofix** [comment]
Na bespreking hiervan, oa. met @timveld: het label (de `span`) zou een id moeten krijgen, en de progressbar zelf moet daar met een `aria-labeledby` naar verwijzen. Idealiter staat dan in dat label nog wel iets zinvols, namelijk wat er dan (in dit geval) nog ongeveer 4 minuten duurt.

**Comment by gofix** [comment]
Omdat ook de progress-indicator een role="progressbar" heeft, zou ik die graag in een moeite door ook willen corrigeren. In dat component loop ik tegen een `aria-valuetext` aan, waar nu (net als in het label) een tekst als "Even geduld aub" in komt. Ik zou willen voorstellen `aria-valuetext` weg te laten, want, lees ik op https://www.digitala11y.com/aria-valuetext-properties/

> If the aria-valuetext attribute is set, authors SHOULD also set the aria-valuenow attribute, unless that value is unknown (for example, on an indeterminate progressbar).
> Authors SHOULD only set the aria-valuetext attribute when the rendered value cannot be meaningfully represented as a number.

Een progress-indicator (de spinner, waarom heet die niet eigenlijk gewoon progress-spinner, maar dat terzijde) is indeterminate, dus heeft geen aria-valuenow, en derhalve lijkt het plaatsen van aria-valuetext onjuist.

**Comment by gofix** [comment]
Er zijn nog issues met de **progress indicator** (de spinner, voor indeterminate doeleinden):
Oud: https://core.dso-toolkit.nl/26.0.0/iframe.html?id=progress-indicator--medium&args=&viewMode=story
Nieuwe: https://core.dso-toolkit.nl/_1211-progressbar-accessible-name/iframe.html?id=progress-indicator--medium&args=&viewMode=story

Oude situatie : had role="progressbar" op het component zelf, en daar zat ook aria-valuetext op, die nergens op sloeg.
Nieuwe situatie : omdat we met aria-labelledby naar het label willen verwijzen, maar het label in het shadow dom leeft, is zowel de role="progress" als de aria-labelledby naar het shadow dom verhuisd. Maar... nu lijken NVDA en JAWS het ding niet meer als voortgangsindicator te herkennen...

**Comment by gofix** [comment]
We ervaren dus nog echt wel rare verschillen tussen hoe screenreaders met deze werkvormen omgaan, maar theoretisch lijken de componenten wel in orde. Met @timveld geconcludeerd dat we het er niet op laten hangen.

**Comment by timveld** [comment]
@gofix Klopt, dit staat wel op mijn lijstje om nog eens verder naar te kijken maar ik weet nog niet wat we er aan zouden kunnen doen. Zou ook een bug in Firefox of NVDA kunnen zijn want op mag taat het kennelijk wel goed.
